### ISSUE TITLE
 Unknown mode: [object Object]

### ISSUE THREAD

**Issue by vikash20186** [kucherenko/jscpd#30]
Hi,

I am getting below error. It used to work fine before. Any help would be appreciated.

My config file
#.cpd.yaml
path:
    path of the source
languages:
  javascript
  htmlmixed   # html mixed source like knockout.js templates
exclude:
  "**/*.min.js"
  "**/*.mm.js"
reporter: json

Error

C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\node_modules\codemirror\
addon\runmode\runmode.node.js:100
  if (!mfactory) throw new Error("Unknown mode: " + spec);
                       ^
Error: Unknown mode: [object Object]
  at Object.exports.getMode (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\j
scpd\node_modules\codemirror\addon\runmode\runmode.node.js:100:24)
  at evalmachine.<anonymous>:19:28
  at Object.exports.getMode (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\j
scpd\node_modules\codemirror\addon\runmode\runmode.node.js:101:10)
  at Object.exports.runMode (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\j
scpd\node_modules\codemirror\addon\runmode\runmode.node.js:106:22)
  at TokenizerCodeMirror.tokenize (C:\Users\c0agav1\AppData\Roaming\npm\node_mod
ules\jscpd\src\tokenizer\TokenizerCodeMirror.coffee:41:16)
  at TokenizerCodeMirror.tokenize (C:\Users\c0agav1\AppData\Roaming\npm\node_mod
ules\jscpd\src\tokenizer\TokenizerCodeMirror.coffee:1:1)
  at Strategy.detect (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\sr
c\strategy.coffee:29:47)
  at Detector.start (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\src
\detector.coffee:10:15)
  at jscpd.run (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\src\jscp
d.coffee:106:26)
  at Object.<anonymous> (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd
\src\cli\cli.coffee:34:10)
  at after (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\node_modules
\cli\cli.js:1009:18)
  at Object.cli.main (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\no
de_modules\cli\cli.js:1014:9)
  at Object.<anonymous> (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd
\src\cli\cli.coffee:27:5)
  at Object.<anonymous> (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd
\src\cli\cli.coffee:1:1)
  at Module._compile (module.js:456:26)
  at Object.loadFile (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\no
de_modules\coffee-script\lib\coffee-script\register.js:16:19)
  at Module.load (C:\Users\c0agav1\AppData\Roaming\npm\node_modules\jscpd\node_m

**Comment by kucherenko** [comment]
Try to use language html, not htmlmixed

**Comment by skyl** [comment]
I get a similar traceback as above when trying to run against js and html templates:

```
        options = {
            path: path,
            reporter: 'json',
            files: [
                '**/*.js', '**/*.html'
            ],
            exclude: [
                '**/*main.js', '**/*build.js', '**/webspec/**'
            ],
            output: '.cpReport.json'
        };
        jscpd.prototype.run(options);
```

--->

```
Error: Unknown mode: [object Object]
  at Object.exports.getMode (node_modules/jscpd/node_modules/codemirror/addon/runmode/runmode.node.js:100:24)
  at evalmachine.<anonymous>:19:28
  at Object.exports.getMode (node_modules/codemirror/addon/runmode/runmode.node.js:101:10)
  at Object.exports.runMode (node_modules/codemirror/addon/runmode/runmode.node.js:106:22)
  at TokenizerCodeMirror.tokenize (node_modules/jscpd/src/tokenizer/TokenizerCodeMirror.coffee:41:16)
  at TokenizerCodeMirror.tokenize (node_modules/jscpd/src/tokenizer/TokenizerCodeMirror.coffee:1:1)
  at Strategy.detect (node_modules/jscpd/src/strategy.coffee:29:47)
  at Detector.start (node_modules/jscpd/src/detector.coffee:10:15)
  at jscpd.run (node_modules/jscpd/src/jscpd.coffee:105:26)
```

With `languages=html`, no code is found.

```
jscpd --path src --files "**/*.html" -l 100 --languages=html
...
100.00% (0 lines) duplicated lines out of 0 total lines of code.
```

With `htmlmixed`, I get a slightly different error:

```
> jscpd --path src --files "**/*.html" -l 100 --languages=htmlmixed
info:    jscpd - copy/paste detector for programming source code, developed by Andrey Kucherenko
warn:    File src/.cpd.yaml not found in current directory, or it is broken
warn:    File src/.cpd.yml not found in current directory, or it is broken
info:    Files search time: durationMs=550
info:    Scanning 869 files for duplicates...

evalmachine.<anonymous>:54
      state.localState = cssMode.startState(htmlMode.indent(state.htmlState, "
                                 ^
TypeError: Object [object Object] has no method 'startState'
    at Object.html (evalmachine.<anonymous>:54:34)
    at Object.token (evalmachine.<anonymous>:101:20)
    at Object.exports.runMode (node_modules/jscpd/node_modules/codemirror/addon/runmode/runmode.node.js:171:24)
    at TokenizerCodeMirror.tokenize (node_modules/jscpd/src/tokenizer/TokenizerCodeMirror.coffee:66:18)
    at TokenizerCodeMirror.tokenize (node_modules/jscpd/src/tokenizer/TokenizerCodeMirror.coffee:3:59)
    at Strategy.detect (node_modules/jscpd/src/strategy.coffee:36:23)
    at Detector.start (node_modules/jscpd/src/detector.coffee:25:23)
    at jscpd.run (node_modules/jscpd/src/jscpd.coffee:129:28)
    at Object.<anonymous> (node_modules/jscpd/src/cli/cli.coffee:35:21)
    at after (node_modules/jscpd/node_modules/cli/cli.js:1057:18)
```

**Comment by kucherenko** [comment]
Fixed in 0.5.0
